src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/QualityGuidelines/CSharpAddMissingInterpolationToken.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpAddMissingInterpolationToken : AbstractAddMissingInterpolationTokenAnalyzer
+    {
+        private protected override bool ShouldReport(ILiteralOperation operation)
+        {
+            var annotation = new SyntaxAnnotation();
+            if (SyntaxFactory.ParseExpression("$" + operation.Syntax.ToString(), options: operation.Syntax.SyntaxTree.Options).WithAdditionalAnnotations(annotation)
+                is not InterpolatedStringExpressionSyntax dummyNode)
+            {
+                return false;
+            }
+
+            var root = operation.Syntax.SyntaxTree.GetRoot();
+            root = root.ReplaceNode(operation.Syntax, dummyNode);
+            dummyNode = (InterpolatedStringExpressionSyntax)root.GetAnnotatedNodes(annotation).Single();
+            if (!operation.SemanticModel.TryGetSpeculativeSemanticModel(operation.Syntax.SpanStart, dummyNode.FirstAncestorOrSelf<StatementSyntax>(), out var model))
+            {
+                return false;
+            }
+
+            var interpolations = dummyNode.Contents.OfType<InterpolationSyntax>();
+            bool hasNonConstantInterpolation = false;
+            foreach (var interpolation in interpolations)
+            {
+                if (interpolation.Expression is LiteralExpressionSyntax)
+                {
+                    continue;
+                }
+
+                hasNonConstantInterpolation = true;
+                if (model.GetSymbolInfo(interpolation.Expression).Symbol is null)
+                {
+                    return false;
+                }
+            }
+
+            return hasNonConstantInterpolation;
```

Comment:
I missed that in the previous review but I don't see the point of this variable, it's always true at this point. I would simply return true here and get rid of the varaible.

Reply:
@Evangelink It should be false in cases like `"Hello world"`, or `"Hello, {0}"`. (ie, no interpolations or all interpolation are literals)

Reply:
Ahh sorry! With the comments in between I thought that the closing curly brace was for the if not the loop... Feel free to resolve.